src/Microsoft.Azure.WebJobs.Host/Singleton/SingletonManager.cs

```diff
@@ -166,7 +166,7 @@ public async virtual Task ReleaseLockAsync(RenewableLockHandle handle, Cancellat
                 await handle.LeaseRenewalTimer.StopAsync(cancellationToken);
             }
 
-            await _lockManager.ReleaseLockAsync(handle.InnerLock, cancellationToken);
+            await _lockManager.ReleaseLockAsync(handle.InnerLock, CancellationToken.None);
```

Comment:
I'm wondering if this is the right layer to make this fix. From the standpoint of the isolated SingletonManager class, it makes sense that its operations take a token and allow the caller to cancel those operations. It's up to the caller to decide whether they want to support cancellation, or to pass CancellationToken.None in on the call.

So I think we should look up the call stack here - perhaps it's at the point where we're calling this method that we need to pass None?

Reply:
@mathewc that's a good point. A layer up from here is [SingletonLock](https://github.com/Azure/azure-webjobs-sdk/blob/aeb7c0c6a4e3dc72d0528b75fa5e797e453cc6dd/src/Microsoft.Azure.WebJobs.Host/Singleton/SingletonLock.cs#L100) which also doesn't feel like the right place for it; but a layer up from that is where we have an exception and decide to release the lock in [FunctionExecutor](https://github.com/Azure/azure-webjobs-sdk/blob/aeb7c0c6a4e3dc72d0528b75fa5e797e453cc6dd/src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs#L348) and I think that feels like the right point to pass in `CancellationToken.None`. I've updated the code to have that change - let me know what you think!